src/NvGet/Tools/Updater/Extensions/XmlDocumentExtensions.cs

```diff
@@ -16,6 +16,46 @@ namespace NvGet.Tools.Updater.Extensions
 {
 	public static class XmlDocumentExtensions
 	{
+		/// <summary>
+		/// Runs an <see cref="UpdateOperation"/> on Project Properties contained in a <see cref="XmlDocument"/>.
+		/// </summary>
+		/// <param name="document"></param>
+		/// <param name="operation"></param>
+		/// <returns></returns>
+		public static IEnumerable<UpdateOperation> UpdateProjectProperties(
+			this XmlDocument document,
+			UpdateOperation operation,
+			ICollection<(string PropertyName, string PackageId)> projectProperties)
+		{
+			var operations = new List<UpdateOperation>();
+
+			var packageId = operation.PackageId;
+
+			foreach(var prop in projectProperties.Where(x=>x.PackageId==operation.PackageId))
+			{
+				var docProp = document.SelectElements(prop.PropertyName).FirstOrDefault();
```

Comment:
In theory, it's possible there are multiple declarations of the property with different conditions. How will we be handling that? Should we update *all* property references?

Reply:
Sticking with the simple use case of it occurring once